fixing map_rotator component using css3 transformation

lesion opened the pull request:
Ok, in my sparse time I'll try to give back something :)
I needed a way to make leaflet rotate following heading of GPS:
your map_rotator component was perfect but not working, this is a fix !
Don't know if you wanna merge this, feel free to ignore this obviously.
PS. I'll PR OSRMv5 routing component for sure because a lot of work was done by @sweco-semhul
[here](https://github.com/ffwdme/ffwdme.js/compare/master...sweco-semhul:osrm) and this saved a lot of time to me and probably would be helpful to someone else.

ciao

lesion committed:
fixing map_rotator component using css3 transformation

lesion commented on src/components/leaflet/index.js:
```diff
@@ -93,7 +93,7 @@ var Leaflet = Base.extend({
   rotateMarker: function(e) {
     var heading = e.geoposition.coords.heading;
-    heading && this.marker && this.marker.setIconAngle(heading);
+    this.marker && this.marker.setIconAngle(heading);
```
removed heading because could be also 0

fabrik42 replied:
You are right, this was a bug!
The reason why this check was here is because there is no guaranteed `heading` property in a Coordinates object.

>  If speed is 0, heading is NaN. If the device is unable to provide heading information, this value is null.

from https://developer.mozilla.org/en-US/docs/Web/API/Coordinates

My implementation was buggy, but I think we should keep the check, just with a better implementation, like checking for NaN and null explicitly. What do you think?

lesion replied:
you're right, what do you think about something like:

`!isNaN(heading) && heading !== null && this.marker && this.marker.setIconAngle(heading);`

or an explicit `if` is needed in this case ?

fabrik42 replied:
I think an explicit `if` makes it a little bit more readable :)

lesion replied:
take a look now

fabrik42 commented on src/components/map_rotator/index.js:
```diff
@@ -35,8 +37,18 @@ var MapRotator = Base.extend({
+    if(diff>180){
+      heading-=360
+    }
+    else if(diff<-180){
```
minor nitpicks:
- Could you add spaces around the parenthesis of the `if` statement and the binary operators? e.g. `<-` looks like it's an operator on its own.
- Use semicolons to and a statement.
- No need for a new line for the else condition.

It would be cool, if you could adjust your code to the general style of this project.

```
if (diff > 180) {
  heading -= 360;
} else if (diff < -180) {
  heading += 360;
}
```

fabrik42 commented on src/components/map_rotator/index.js:
```diff
@@ -35,8 +37,18 @@ var MapRotator = Base.extend({
 
     var heading = - e.geoposition.coords.heading;
-    heading && this.map.el && this.map.el.animate({ rotate: heading + 'deg' }, 1000, 'ease-in-out');
+    var diff = heading-this.last_heading
```
You dropped the `this.map.el` check. Is there a special reason why you did this?

fabrik42 commented:
Hi!
This looks very good, thank you!
I just added some comments for finishing touches on the code style and some checks - afterwards I would be happy to merge it.

Let me know your thoughts on my comments. :)

Have a nice weekend!

Chris

lesion committed:
Check if heading is available:
If speed is 0, heading is NaN. If the device is unable to provide heading information, this value is null.
from https://developer.mozilla.org/en-US/docs/Web/API/Coordinates

lesion committed:
adjust PR code style to project style

lesion committed:
cleaner heading checking

sweco-semhul commented:
@lesion that would be great! 
My OSRM routing support is pretty much working ok. But got into some problems when running the example, which I ment to fix before I made a PR ... after that I haven´t got back to it.

fabrik42 commented:
@lesion Very cool, thank you!

@sweco-semhul @lesion If you need any assistance let me know, I'm happy to help!
